packages/api/tests/endpoints/test-rules.js

```diff
@@ -580,51 +579,64 @@ test.serial('POST does not write to DynamoDB or RDS if writing to DynamoDB fails
   t.false(await ruleModel.exists(newRule.name));
 });
 
-test('PUT replaces a rule', async (t) => {
+test.serial('PUT replaces a rule', async (t) => {
   const { dbClient } = t.context;
-  const expectedRule = {
-    ...omit(testRule, ['queueUrl', 'provider', 'collection']),
+  const putTestRule = { ...testRule, name: randomId('testRule') };
+  t.truthy(putTestRule.queueUrl);
+
+  const updateRule = {
+    ...omit(putTestRule, ['queueUrl', 'provider', 'collection']),
     state: 'ENABLED',
   };
 
-  const translatedRule = await translateApiRuleToPostgresRule(testRule, dbClient);
-  await dbClient(tableNames.rules).insert(translatedRule);
-
-  const dbRecord = await dbClient.select()
-    .from(tableNames.rules)
-    .where({ name: expectedRule.name })
-    .first();
-
-  // Make sure testRule contains values for the properties we omitted from
-  // expectedRule to confirm that after we replace (PUT) the rule those
-  // properties are dropped from the stored rule.
-  t.truthy(testRule.queueUrl);
+  await request(app)
+    .post('/rules')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(putTestRule)
+    .expect(200);
```

Comment:
I kind of dislike using other endpoints to test the behavior of the PUT endpoint. It means that this test isn't really isolated to testing PUT endpoint behavior. I realize that using model methods is also inherently testing them as well, but that feels like a cleaner separation of concerns than using other endpoints directly

Reply:
👍🏻 entirely fiar.

Reply:
On second thought, this warrants a bit more thought - the endpoint logic is the location we chose for the rule write/read logic to live for dual database write logic, not the model.    

If the intent of this test is to test that the rule is replaced in our current read database.... as originally written, this test
 
- Writes a record to the db, then selects it back from the DB..... 
- Then runs a database `.put` using the code under test
- *Using the API* checks what is returned, reading from *Dynamodb* 
- Using the database selects a row and checks it. 

I don't believe this is correct - given the current code, we need to insert intially into *both* dynamo and RDS -- either using the code in the endpoint *or replicating it here directly* 

I get not wanting to use `request.put`, but I'm not sure that duplicating the current PUT write logic that's encapsulated in the endpoint code is right either - if anything it feels a bit like the decision not to abstract that might have been a mistake. 

I'll update this test as described, but under some duress.

Reply:
Ignore the above, I missed the database write in the test setup at first glace.     Will re-comment.

Reply:
This is particuarly gross, as the interface for the dynamo model is: 

  async create(item, createdAt) {

which ignores the value in the item and explicitly forces you to pass it in (for testing?), this took me longer than I care to admit to debug. 

In either case, updated, and it *is* a better test now in that it tests the dynamo record explicitly.    

4961b86f7